src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -576,6 +576,9 @@ event::ptr primitive_inst::realloc_if_needed() {
         if (can_be_optimized()) {
             _max_output_layout_count = _deps[0].first->_max_output_layout_count;
             GPU_DEBUG_PROFILED_STAGE_MEMALLOC_INFO("can_be_optimized");
+            if (_outputs[0] && !_outputs[0]->get_layout().identical(updated_layout)) {
+                _outputs[0] = _network.get_engine().reinterpret_buffer(*_outputs[0], updated_layout);
+            }
```

Comment:
Can I ask what was the primitive being skipped? 
Actually the original intention is to reinterpret buffer is to be done by each primitive_inst::update_output_memory at on_execute().
e.g., if permute is opt out, we are reinterpret buffer at permute_inst::update_output_memory() called from permute_inst::on_execute(). 
So I think if we needed to do that reinterpret here, something is missing for the original flow.

Reply:
That primitive is reorder. I agree with you and have changed the location to reinterpret in reorder.